components/ws-manager-bridge/src/bridge.ts

```diff
@@ -504,11 +510,21 @@ export class WorkspaceManagerBridge implements Disposable {
             const actuallyRunningInstances = await client.getWorkspaces(ctx, new GetWorkspacesRequest());
             actuallyRunningInstances.getStatusList().forEach((s) => runningInstancesIdx.delete(s.getId()));
 
+            // runningInstancesIdx only contains instances that ws-manager is not aware of
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
-                if (instance.status.phase !== "running") {
+                const phase = instance.status.phase;
+
+                if (
+                    phase === "pending" ||
+                    phase === "creating" ||
+                    phase === "initializing" ||
+                    (phase === "stopping" &&
+                        instance.stoppingTime &&
+                        durationLongerThanSeconds(Date.parse(instance.stoppingTime), 10))
+                ) {
                     log.debug({ instanceId }, "Skipping instance", {
```

Comment:
I think we want to stop these instances as well as when `phase === 'running'`. So maybe you want to negate 😅 the logic or just move the lines 534-539 up into this if block (and also include phase===running).

Reply:
My brain 🤦🏻‍♀️🤦🏻‍♀️🤦🏻‍♀️